File: backend/internal/entity/ebook.go
package entity

import (
	"time"

	"github.com/google/uuid"
)

type Ebook struct {
	EbookID         uuid.UUID `json:"ebook_id"`
	Title           string    `json:"title"`
	Author          string    `json:"author"`
	Description     string    `json:"description"`
	Publisher       string    `json:"publisher"`
	PublicationDate time.Time `json:"publication_date"`
	ISBN            string    `json:"isbn"`
	TotalPages      int       `json:"total_pages"`
	IsPublished     bool      `json:"is_published"`
	Category        string    `json:"category"`
	EbookURL        string    `json:"ebook_url"`
	CoverImageURL   string    `json:"cover_image_url"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

func (Ebook) TableName() string {
	return "ebooks"
}

File: web/src/types/ebook.ts
export interface EbookProps {
  ebook_id: string;
  title: string;
  author: string;
  description: string;
  publisher: string;
  publication_date: Date | string;
  isbn: string;
  total_pages: number;
  is_published: boolean;
  category: string;
  ebook_url: string;
  cover_image_url: string;
  created_at: Date | string;
  updated_at: Date | string;
}
